src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,656 @@
+            {
+                invocationReturn = new List<SyntaxNode>
+                {
+                    generator.ReturnStatement(invocation)
+                };
+            }
+            else
+            {
+                invocationReturn = new List<SyntaxNode>
+                {
+                    invocation
+                };
+            }
+
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol);
+
+            var methodDeclaration = generator.MethodDeclaration(newMethod, invocationReturn);
+
+            editor.InsertBefore(methodExpression, methodDeclaration);
+        }
+
+        public static void UpdateExpressionInOriginalFunction(SemanticDocument semanticDocument,
+                                                              TExpressionSyntax expression, SyntaxNode scope,
+                                                              string parameterName, SyntaxEditor editor,
+                                                              bool allOccurrences, CancellationToken cancellationToken)
+        {
+            var generator = SyntaxGenerator.GetGenerator(semanticDocument.Document);
+            var matches = FindMatches(semanticDocument, expression, semanticDocument, scope, allOccurrences, cancellationToken);
+            var parameterNameSyntax = (TIdentifierNameSyntax)generator.IdentifierName(parameterName);
+            // Parenthesize the variable, and go and replace anything we find with it.
+            // NOTE: we do not want elastic trivia as we want to just replace the existing code 
+            // as is, while preserving the trivia there.  We do not want to update it.
+
+            var replacement = generator.AddParentheses(parameterNameSyntax, includeElasticTrivia: false)
+                                         .WithAdditionalAnnotations(Formatter.Annotation);
+
+            var syntaxFacts = semanticDocument.Document.GetRequiredLanguageService<ISyntaxFactsService>();
+
+            foreach (var match in matches)
+            {
+                if (match.Equals(expression))
+                {
+                    if (IsExpressionRemovable(semanticDocument, expression))
+                    {
+                        var localDeclaration = expression.FirstAncestorOrSelf<SyntaxNode>(node => syntaxFacts.IsLocalDeclarationStatement(node));
+                        if (localDeclaration is not null)
+                        {
+                            editor.RemoveNode(localDeclaration);
+                        }
+                    }
+                    else
+                    {
+                        var expressionToken = syntaxFacts.GetIdentifierOfIdentifierName(parameterNameSyntax);
+                        var updatedExpressionToken = expressionToken.WithAdditionalAnnotations(RenameAnnotation.Create());
+                        parameterNameSyntax = parameterNameSyntax.ReplaceToken(expressionToken, updatedExpressionToken);
```

Comment:
`generator.IdentifierName(generator.Identifier(parameterName).WithAdditionalAnnotations(...)`

Reply:
Changed, didn't know you could do that